### PR TITLE
Add controllers

### PR DESCRIPTION
**Pull request by Zeph180**
1. For each URL you should create:
- A route.
- An action in the correct controller.
- A view file.

2. View files should include only placeholders written in plain HTML, e.g:
- "Here is a list of posts for a given user"

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class PostsController < ApplicationController
+  def show
+    @post = Post.find_by(id: params[:id])
+  end
+end
```
Comment (rloterh) [inline app/controllers/posts_controller.rb]: - [ ] `index` action is missing in the _posts_ controller. Kindly ensure that both `index` and `show` actions are present for the posts controller so as to achieve corresponding _views_ similar to the given [wireframe template](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Setup_Controllers_project_v1_1.md).

File: app/views/posts/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,4 @@
+<h1>Post Details</h1>
+<p>Title: </p>
+<p>Content: </p>
+<p>Author: </p>
```
Comment (rloterh) [inline app/views/posts/show.html.erb:4]: - [ ] Similarly, let's add a corresponding _view_ (`# app/views/posts/index.html.erb`) to match the `index` actions in the _posts_ controller.

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Zeph, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there!
🏁 🚦 ⛪ 🚜 🏃 🌴 🏫 🌳  🏘️  

## Highlights 👍  
- Good job so far adding controllers and associated views and routes!
- ReadMe looks good!

## Required Changes ♻️
_Kindly check the inline comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

**You can also consider:**
- Let's improve the look of the ReadMe by removing the circled aspect below:

image

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class PostsController < ApplicationController
+  def show
+    @post = Post.find_by(id: params[:id])
+  end
```
Comment (mikethreels) [inline app/controllers/posts_controller.rb:9]: - [ ] good job adding the post index file in the views, however, as the previous CR already mentioned there should be an index action in the post controller as well. please ensure this is included as well.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Zeph, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- good job following the instructions from the previous CR

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Reem-lab — APPROVED** [review]
# Hi @Zeph180, 👋🏻👋🏻

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights
- [ ] Great job on the functionaliy for set up the controller 💯
- [ ] Professional README 🔥
- [ ] Followed Gitflow 🔥

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**